### PR TITLE
refactor(Modal)!: Deprecating Modal component

### PR DESCRIPTION
**Pull request by thedoomshine**
### Overview

Deprecating Modal component and all uses of it

### PR Checklist

- [X] Related to designs:
- [X] Related to JIRA ticket: [GM-142](https://codecademy.atlassian.net/browse/GM-142)
- [X] I have run this code to verify it works
- [X] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.104-alpha.43c88e.0<br/>@codecademy/gamut-labs@17.1.2-alpha.43c88e.0<br/>@codecademy/gamut@32.18.9-alpha.43c88e.0<br/>@codecademy/styleguide@38.1.2-alpha.43c88e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2d3d5768736252d91e904--gamut-preview.netlify.app>https://60a2d3d5768736252d91e904--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2d3d5768736252d91e904>Deploy Logs</a></p>

File: packages/styleguide/stories/Atoms/Overlay.stories.mdx [hunk 1]
```diff
@@ -5,10 +5,11 @@ import { PropsTable } from '@codecademy/storybook-addon-variance';
...
 
-<Meta
-  title={title}
-  component={Overlay}
-  parameters={{
+    <Meta
+
+title={title}
+component={Overlay}
+parameters={{
```
Comment (codecaaron) [inline packages/styleguide/stories/Atoms/Overlay.stories.mdx:12]: I think the ESlint parser might have mangled this MDX file with find and replace.

I think if you remove https://github.com/Codecademy/client-modules/pull/1763/files#diff-21d737ce8300a17a65a18cfcf8e72a865140d35d7aafbfa293e9a33532c5df4dR9 and safe it locally it should fix itself (if you have the vscode plugins)

**Review by codecaaron — APPROVED** [review]
💯  pending the MDX Parsing fix in one of the stories.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.104-alpha.51ef63.0<br/>@codecademy/gamut-labs@17.1.2-alpha.51ef63.0<br/>@codecademy/gamut@32.18.9-alpha.51ef63.0<br/>@codecademy/styleguide@38.1.2-alpha.51ef63.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2d9e6fd3d98269444201d--gamut-preview.netlify.app>https://60a2d9e6fd3d98269444201d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2d9e6fd3d98269444201d>Deploy Logs</a></p>

File: packages/styleguide/stories/Atoms/Overlay.stories.mdx (hunk 1 above)
Comment (thedoomshine) [reply packages/styleguide/stories/Atoms/Overlay.stories.mdx]: fixed!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.104-alpha.7c9705.0<br/>@codecademy/gamut-labs@17.1.2-alpha.7c9705.0<br/>@codecademy/gamut@32.18.9-alpha.7c9705.0<br/>@codecademy/styleguide@38.1.2-alpha.7c9705.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2db4ffd3d9836db44194e--gamut-preview.netlify.app>https://60a2db4ffd3d9836db44194e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2db4ffd3d9836db44194e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.104-alpha.1f68e7.0<br/>@codecademy/gamut-labs@17.1.2-alpha.1f68e7.0<br/>@codecademy/gamut@32.18.9-alpha.1f68e7.0<br/>@codecademy/styleguide@38.1.2-alpha.1f68e7.0<br/></pre>

**Review by JoshuaKGoldberg — APPROVED** [review]
Woot!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.112-alpha.cf5988.0<br/>@codecademy/gamut-labs@18.1.3-alpha.cf5988.0<br/>@codecademy/gamut@32.20.2-alpha.cf5988.0<br/>@codecademy/styleguide@39.0.5-alpha.cf5988.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60abfda51ea08d0fcef84a89--gamut-preview.netlify.app>https://60abfda51ea08d0fcef84a89--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60abfda51ea08d0fcef84a89>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🔥

**Pull request merged**